src/components/DashboardPage/DashboardPage.tsx

```diff
@@ -0,0 +1,270 @@
+import { createFilterKeys } from '../../utils/hotkeys';
+import { useUrlFilterParams } from '../../hooks/useUrlFilterParams';
+import { useFilterResource } from '../../hooks/useFilterResource';
+import { routes } from '../../hooks/router';
+import { Page, PageContent } from '../Page';
+import { CommonHeader } from '../CommonHeader';
+import { FiltersPanel } from '../FiltersPanel/FiltersPanel';
+import { GoalsGroup } from '../GoalsGroup';
+import { GoalsListContainer } from '../GoalListItem';
+import { PageTitle } from '../PageTitle';
+import { Nullish } from '../../types/void';
+import { trpc } from '../../utils/trpcClient';
+import { FilterById, GoalByIdReturnType, ProjectByIdReturnType } from '../../../trpc/inferredTypes';
+import { ProjectItemStandalone } from '../ProjectListItem';
+
+import { tr } from './DashboardPage.i18n';
+
+const GoalPreview = dynamic(() => import('../GoalPreview/GoalPreview'));
+const ModalOnEvent = dynamic(() => import('../ModalOnEvent'));
+const FilterCreateForm = dynamic(() => import('../FilterCreateForm/FilterCreateForm'));
+const FilterDeleteForm = dynamic(() => import('../FilterDeleteForm/FilterDeleteForm'));
+
+export const DashboardPage = ({ user, ssrTime, locale }: ExternalPageProps) => {
+    const router = useRouter();
+    const [preview, setPreview] = useState<GoalByIdReturnType | null>(null);
+    const { toggleFilterStar } = useFilterResource();
+
+    const utils = trpc.useContext();
+
+    const presetData = trpc.filter.getById.useQuery(router.query.filter as string, { enabled: !!router.query.filter });
+
+    const {
+        currentPreset,
+        queryState,
+        queryString,
+        setPriorityFilter,
+        setStateFilter,
+        setTagsFilter,
+        setTagsFilterOutside,
+        setEstimateFilter,
+        setIssuerFilter,
+        setOwnerFilter,
+        setParticipantFilter,
+        setProjectFilter,
+        setStarredFilter,
+        setWatchingFilter,
+        setSortFilter,
+        setFulltextFilter,
+        resetQueryState,
+        setPreset,
+    } = useUrlFilterParams({
+        preset: presetData?.data,
+    });
+
+    const { data, isLoading } = trpc.goal.getUserGoals.useQuery(queryState, {
+        keepPreviousData: true,
+        staleTime: refreshInterval,
+    });
+
+    const goals = data?.goals;
+    const meta = data?.meta;
+    const userFilters = trpc.filter.getUserFilters.useQuery();
+    const shadowPreset = userFilters.data?.filter((f) => f.params === queryString)[0];
+
+    const groupsMap =
+        // eslint-disable-next-line no-spaced-func
+        (goals as NonNullable<GoalByIdReturnType>[])?.reduce<{
+            [key: string]: {
+                project?: ProjectByIdReturnType | null;
+                goals: NonNullable<GoalByIdReturnType>[];
+            };
+        }>((r, g) => {
+            const k = g.projectId;
+
+            if (k) {
+                if (!r[k]) {
+                    r[k] = {
```

Comment:
That doesn't look good, you probably should use longer names

Reply:
https://github.com/taskany-inc/issues/issues/1095